src/application-manager.coffee

```diff
@@ -200,6 +187,19 @@ module.exports = class ApplicationManager extends EventEmitter
 		.then (interval) =>
 			@images.appUpdatePollInterval = interval
 			@images.cleanupDatabase()
+		.then =>
+			# Rather than relying on removing out of data database entries when we're no
+			# longer using them, set a task that runs periodically to clear out the database
+			# This has the advantage that if for some reason a container is removed while the
+			# supervisor is down, we won't have zombie entries in the db
+			setInterval =>
```

Comment:
This would be nicer as
```coffee
fn = =>
	@docker.listContainers(all: true).then (containers) =>
		@logger.clearOutOfDateDBLogs(_.map(containers, 'Id'))
setInterval(fn, 1000 * 60 * 60 * 24)
fn()
```